### DISCUSSION TITLE
Support Gradle RepositoryContentDescriptor

### DISCUSSION THREAD

**Discussion by marverix** [renovatebot/renovate#32256] [Suggest an Idea]
Gradle has [RepositoryContentDescriptor](https://docs.gradle.org/current/kotlin-dsl/gradle/org.gradle.api.artifacts.repositories/-repository-content-descriptor/index.html) which allows you to configure which packages, modules, groups, etc should be excluded/included . One can use it to tell Gradle to (not) look into given repository if given package is processed.

Example, `settings.gradle`:
```groovy
dependencyResolutionManagement {
    repositories {
        mavenCentral {
            content {
                excludeGroupByRegex("company.*")
            }
        }
        maven {
            url 'https://my.custom.registry/maven/'
        }
    }
}
```

This could be translated to Renovate's:

```
"packageRules": [
  {
    "matchCategories": ["java"],
    "registryUrls": ["https://my.custom.registry/maven/"]
  },
  {
    "matchCategories": ["java"],
    "matchDepNames": ["!/company.*"],
    "registryUrls": ["https://repo.maven.apache.org/maven2/"]
  }
]
```

**Comment by Churro** [comment]
This feature request is related to https://github.com/renovatebot/renovate/issues/14208 but should probably be tracked separately anyway, due to the different patterns that need addition. renovate's gradle manager currently lacks support for [RepositoryContentDescriptor](https://docs.gradle.org/current/kotlin-dsl/gradle/org.gradle.api.artifacts.repositories/-repository-content-descriptor/index.html). The proposal to translate parsed statements into package rules seems appealing, but by the time parsing happens, it is already too late to add package rules.

Note that `excludeGroupByRegex` is just one possible flavor but there's also `include{Group,Module,Version}`, then `include{GroupByRegex, ModuleByRegex, VersionByRegex}`, and the corresponding `exclude` counterparts that would each need separate parsing and handling.

**Reply by rarkins** [reply]
Does it need a reproduction before converting to an Issue?

**Reply by Churro** [reply]
Would be great to test any implementation against it but I wouldn't expect it to cover all possible flavors (as mentioned before).

**Comment by Churro** [comment]
-> PR https://github.com/renovatebot/renovate/pull/33692
